units/provisioning_setup_host.go

```diff
@@ -306,6 +281,84 @@ func (j *setupHostJob) setupJasper(ctx context.Context) error {
 		"job":     j.ID(),
 		"distro":  j.host.Distro.Id,
 	})
+
+	return nil
+}
+
+// putJasperCredentials creates Jasper credentials for the host and puts the
+// credentials file on the host.
+func (j *setupHostJob) putJasperCredentials(ctx context.Context, fileName string, sshOptions []string) error {
+	creds, err := j.host.GenerateJasperCredentials(ctx, j.env)
+	if err != nil {
+		return errors.Wrap(err, "could not generate Jasper credentials for host")
+	}
+
+	hostInfo, err := j.host.GetSSHInfo()
+	if err != nil {
+		return errors.Wrap(err, "could not get SSH info for host")
+	}
+	exportedCreds, err := creds.Export()
+	if err != nil {
+		return errors.Wrap(err, "could not export Jasper credentials")
+	}
+
+	file, err := ioutil.TempFile("", fileName)
+	if err != nil {
+		return errors.Wrap(err, "error creating temporary script file")
+	}
+	if err = os.Chmod(file.Name(), 0644); err != nil {
+		return errors.Wrap(err, "error setting file permissions")
+	}
+	defer func() {
+		errMsg := message.Fields{
+			"job":       j.ID(),
+			"operation": "cleaning up after copying credentials",
+			"file":      file.Name(),
+			"distro":    j.host.Distro.Id,
+			"host":      j.host.Id,
+		}
+		grip.Error(message.WrapError(file.Close(), errMsg))
+		grip.Error(message.WrapError(os.Remove(file.Name()), errMsg))
+	}()
+
+	if _, err = io.WriteString(file, string(exportedCreds)); err != nil {
+		return errors.Wrap(err, "error writing local credentials")
+	}
+
+	scpCmdOut := &util.CappedWriter{
+		Buffer:   &bytes.Buffer{},
+		MaxBytes: 1024 * 1024,
+	}
+	scpArgs := buildScpCommand(file.Name(), fileName, hostInfo, hostInfo.User, sshOptions)
```

Comment:
we should make a note to ourselves that we should be able to move this kind of thing to jasper after MAKE-824 (which you could take in the next couple of sprints if you want.)

Reply:
This is mostly the same as what the `(*setupHostJob).copyScript()` already does, but I'll refactor it in [EVG-6347](https://jira.mongodb.org/browse/EVG-6347).